spec/api/channel_spec.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'discordrb'
+
+describe Discordrb::API::Channel do
+  let(:token) { double('token') }
+  let(:channel_id) { instance_double(String, 'channel_id') }
+
+  before do
+    allow(token).to receive(:to_s).and_return('bot_token')
+    allow(channel_id).to receive(:to_s).and_return('channel_id')
+  end
+
+  describe '.get_reactions' do
+    let(:message_id) { double('message_id') }
+    let(:before_id) { double('before_id') }
+    let(:after_id) { double('after_id') }
+
+    before do
+      allow(message_id).to receive(:to_s).and_return('message_id')
+      allow(before_id).to receive(:to_s).and_return('before_id')
+      allow(after_id).to receive(:to_s).and_return('after_id')
+
+      allow(Discordrb::API).to receive(:request)
+        .with(anything, channel_id, :get, kind_of(String), any_args)
+    end
+
+    it 'sends requests' do
+      Discordrb::API::Channel.get_reactions(token, channel_id, message_id, 'test', before_id, after_id, 27)
+      expect(Discordrb::API).to have_received(:request)
```

Comment:
while `have_received` is valid spec'ing known as using spies, I'd prefer we stay consistent and put the expectations before the call